### PR TITLE
PXP-4931 Handle WTS's IDP parameter

### PR DESCRIPTION
**Pull request by paulineribeyre**
Jira Ticket: [PXP-4931](https://ctds-planx.atlassian.net/browse/PXP-4931)

### New Features
- Add optional "wtsIDP" argument to the Gen3Fuse CLI, to be used with "wtsURL"

### Improvements
- The Gen3Fuse sidecar can now handle mounting to the workspace files from multiple Data Commons

### CODE REVIEW

File: sidecarDockerrun.sh [hunk 1]
```diff
@@ -13,32 +18,76 @@ cleanup() {
...
+            echo ${row} | base64 -d | jq -r ${1}
+        }
+        # if user is connected, add IDP info to the lists
+        if [ `_jq '.refresh_token_expiration'` != "null" ]; then
+            IDPS+=( `_jq '.idp'` )
+            BASE_URLS+=( `_jq '.base_url'` )
+        fi
+    done
+
+    for i in "${!IDPS[@]}"; do
```
Comment (frickjack) [inline sidecarDockerrun.sh]: What is the `!` for?

File: sidecarDockerrun.sh [hunk 2]
```diff
@@ -13,32 +18,76 @@ cleanup() {
...
-        resp=`curl http://manifestservice-service.$NAMESPACE/ -H "Authorization: bearer $TOKEN_JSON" 2>/dev/null`
+
+    EXTERNAL_OIDC=`curl http://workspace-token-service.$NAMESPACE/external_oidc/ -H "Authorization: bearer ${TOKEN_JSON['default']}" 2>/dev/null | jq -r '.providers'`
+
+    # list of IDPs to get manifests from.
+    # only select IDPs the user is logged into
+    IDPS=( "default" )
+    BASE_URLS=( "https://$HOSTNAME" )
+    for row in $(echo "${EXTERNAL_OIDC}" | jq -r '.[] | @base64'); do
+        _jq() {
```
Comment (frickjack) [inline sidecarDockerrun.sh]: Pull the function out of the loop, and make it take 2 parameters

File: sidecarDockerrun.sh [hunk 3]
```diff
@@ -13,32 +18,76 @@ cleanup() {
...
+            echo ${row} | base64 -d | jq -r ${1}
+        }
+        # if user is connected, add IDP info to the lists
+        if [ `_jq '.refresh_token_expiration'` != "null" ]; then
```
Comment (frickjack) [inline sidecarDockerrun.sh]: Use `$()` instead of back-tick

File: sidecarDockerrun.sh [hunk 4]
```diff
@@ -13,32 +18,76 @@ cleanup() {
...
-    if [ $(df /data/man* | sed '1d' | wc -l) -lt 7 ]; then # remove header line, and also consider the existence of manifest snyc log
-        resp=`curl http://manifestservice-service.$NAMESPACE/ -H "Authorization: bearer $TOKEN_JSON" 2>/dev/null`
+
+    EXTERNAL_OIDC=`curl http://workspace-token-service.$NAMESPACE/external_oidc/ -H "Authorization: bearer ${TOKEN_JSON['default']}" 2>/dev/null | jq -r '.providers'`
+
+    # list of IDPs to get manifests from.
+    # only select IDPs the user is logged into
+    IDPS=( "default" )
+    BASE_URLS=( "https://$HOSTNAME" )
+    for row in $(echo "${EXTERNAL_OIDC}" | jq -r '.[] | @base64'); do
```
Comment (frickjack) [inline sidecarDockerrun.sh]: You can do 
```
for row in $(jq -r '.[] | @base64' <<<"$EXTERNAL_OIDC"); do
...
```

File: sidecarDockerrun.sh [hunk 5]
```diff
@@ -13,32 +18,76 @@ cleanup() {
...
 trap cleanup SIGTERM
+
+declare -A TOKEN_JSON  # requires Bash 4
+TOKEN_JSON['default']=`curl http://workspace-token-service.$NAMESPACE/token/?idp=default 2>/dev/null | jq -r '.token'`
+
 while true; do
-    if [ $(df /data/man* | sed '1d' | wc -l) -lt 7 ]; then # remove header line, and also consider the existence of manifest snyc log
-        resp=`curl http://manifestservice-service.$NAMESPACE/ -H "Authorization: bearer $TOKEN_JSON" 2>/dev/null`
+
+    EXTERNAL_OIDC=`curl http://workspace-token-service.$NAMESPACE/external_oidc/ -H "Authorization: bearer ${TOKEN_JSON['default']}" 2>/dev/null | jq -r '.providers'`
```
Comment (frickjack) [inline sidecarDockerrun.sh]: Maybe just add a flag to wts?
```
.../external_oidc/?unexpired=true
```

File: sidecarDockerrun.sh (hunk 1 above)
Comment (paulineribeyre) [reply sidecarDockerrun.sh]: `${L[@]}` iterates over the list values and `${!L[@]}` iterates over indices

**Pull request merged**